server/src/websocket/ws-connection.js

```diff
@@ -0,0 +1,22 @@
+const _ = require('lodash')
+
+// Contains methods related to a single connection
+// Binded to WebsocketServer to access 'this'
+class WsConnection {
+  static connection (connection) {
+    this.connections.push(connection)
+    connection.on('close', WsConnection.close.bind(this, connection))
+    connection.on('error', WsConnection.error.bind(this, connection))
+  }
+
+  static close (connection) {
+    _.remove(this.connections, (x) => x === connection)
+  }
+
+  static error (connection, err) {
```

Comment:
Is the connection automatically closed on error? If yes, shouldn't you remove the connection of `this.connections`?

Reply:
I closed it manually on error